src/panels/lovelace/cards/hui-todo-list-card.ts

```diff
@@ -439,10 +438,36 @@ export class HuiTodoListCard
   }
 
   private async _moveItem(oldIndex, newIndex) {
-    const item = this._getUncheckedItems(this._items)[oldIndex];
-    await moveItem(this.hass!, this._entityId!, item.uid!, newIndex).finally(
-      () => this._fetchData()
-    );
+    const uncheckedItems = this._getUncheckedItems(this._items);
+    const item = uncheckedItems[oldIndex];
+    let prevItem: TodoItem | undefined;
+    if (newIndex > 0) {
+      if (newIndex < oldIndex) {
+        prevItem = uncheckedItems[newIndex - 1];
+      } else {
+        prevItem = uncheckedItems[newIndex];
+      }
+    }
+
+    // Optimistic change
+    const itemIndex = this._items!.findIndex((itm) => itm.uid === item.uid);
+    this._items!.splice(itemIndex, 1);
+    if (newIndex === 0) {
+      this._items!.unshift(item);
+    } else {
+      const prevIndex = this._items!.findIndex(
+        (itm) => itm.uid === prevItem!.uid
+      );
+      this._items!.splice(prevIndex + 1, 0, item);
+    }
+    this._items = [...this._items!];
```

Comment:
Is this just making another copy? Seems good just unclear to me as a novice why this is needed.

Reply:
It is to force lit to update, otherwise it thinks `_items` is not changed.

Reply:
You can also do this right?
```ts
this.requestUpdate("_items");
```

Reply:
> You can also do this right?
> 
> ```ts
> this.requestUpdate("_items");
> ```

Not sure if that works with repeat

Reply:
Nope, it doesn't